src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -2838,22 +2853,32 @@ void returnAfter(MethodInformation methodInfo) {
 
         if (methodInfo.returnType == void.class) {
             if (methodInfo.allocator || methodInfo.arrayAllocator) {
-                out.println(indent + "jlong rcapacity = " + (methodInfo.arrayAllocator ? "arg0;" : "1;"));
                 boolean noDeallocator = methodInfo.cls == Pointer.class ||
                         methodInfo.cls.isAnnotationPresent(NoDeallocator.class) ||
                         methodInfo.method.isAnnotationPresent(NoDeallocator.class);
-                out.print(indent + "JavaCPP_initPointer(env, obj, rptr, rcapacity, rptr, ");
-                if (noDeallocator) {
-                    out.println("NULL);");
-                } else if (methodInfo.arrayAllocator) {
-                    out.println("&JavaCPP_" + mangle(methodInfo.cls.getName()) + "_deallocateArray);");
-                    arrayDeallocators.index(methodInfo.cls);
+                if (adapterInfo != null) {
+                    out.println(indent + typeName[0] + " rptr = radapter;");
+                    out.println(indent + "jlong rcapacity = (jlong)radapter.size;");
+                    out.println(indent + "void* rowner = radapter.owner;");
+                    if (noDeallocator)
+                        out.println(indent + "void (*deallocator)(void*) = 0;");
+                    else
+                        out.println(indent + "void (*deallocator)(void*) = rowner != NULL ? &" + adapterInfo.name + "::deallocate : 0;");
+                    out.println(indent + "JavaCPP_initPointer(env, obj, rptr, rcapacity, rowner, deallocator);");
                 } else {
-                    out.println("&JavaCPP_" + mangle(methodInfo.cls.getName()) + "_deallocate);");
-                    deallocators.index(methodInfo.cls);
+                    out.println(indent + "jlong rcapacity = " + (methodInfo.arrayAllocator ? "arg0;" : "1;"));
+                    out.print(indent + "JavaCPP_initPointer(env, obj, rptr, rcapacity, rptr, ");
+                    if (noDeallocator) {
+                        out.println("NULL);");
+                    } else if (methodInfo.arrayAllocator) {
+                        out.println("&JavaCPP_" + mangle(methodInfo.cls.getName()) + "_deallocateArray);");
+                        arrayDeallocators.index(methodInfo.cls);
+                    } else {
+                        out.println("&JavaCPP_" + mangle(methodInfo.cls.getName()) + "_deallocate);");
+                        deallocators.index(methodInfo.cls);
+                    }
                 }
                 if (virtualFunctions.containsKey(methodInfo.cls)) {
-                    typeName = cppTypeName(methodInfo.cls);
```

Comment:
Is there a reason to remove that line here?

Reply:
I guess we could also move that line up a bit, and revert the lines at the beginning of this method too?

Reply:
This assignment is now done line 2804

Reply:
We need the type name for initializing the adapterInfo. That's why typeName is set at the beginning of the method for constructors too.

Reply:
Right, I realize that, but if we can move that line just after `if (methodInfo.allocator || methodInfo.arrayAllocator)` we can revert the changes for its initial assignment, right?

Reply:
`adapterInfo` is initialized above, line 2809. We need the correct valueTypeName at this point. The initial assignment for `typeName` doesn't consider the `methodInfo.allocator` case. Or do I miss something ?

Something that you can double check is the case `methodInfo.arrayAllocator`. I didn't test this code path.